frontend/src/metabase/collections/components/CollectionSidesheet/CollectionSidesheet.tsx

```diff
@@ -0,0 +1,134 @@
+  collection: Collection;
+  onUpdateCollection: (entity: Collection, values: Partial<Collection>) => void;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  useMount(() => {
+    // this component is not rendered until it is "open"
+    // but we want to set isOpen after it mounts to get
+    // pretty animations
+    setIsOpen(true);
+  });
+
+  const iconProps = useMemo(() => {
+    const icon = PLUGIN_COLLECTIONS.getIcon({
+      ...collection,
+      model: "collection",
+    } as ObjectWithModel);
+    if (icon.color) {
+      icon.color = color(icon.color);
+    }
+    return icon;
+  }, [collection]);
+
+  const handleChangeDescription = useCallback(
+    (description: string) => {
+      onUpdateCollection(collection, { description: description || null });
+    },
+    [collection, onUpdateCollection],
+  );
+
+  return (
+    <Sidesheet
+      title={t`Info`}
+      isOpen={isOpen}
+      data-testid="collection-sidesheet"
+      size="md"
+      onClose={onClose}
+    >
+      <Stack spacing="lg">
+        <SidesheetCard pb="md">
+          <Stack spacing="md">
+            <Stack spacing="xs">
+              <Title lh={1} size="sm" color="text-light">
+                {t`Description`}
+              </Title>
+              <EditableDescription
+                collection={collection}
+                handleChangeDescription={handleChangeDescription}
+              />
+            </Stack>
+            {collection.authority_level === "official" && (
+              <Group noWrap spacing="sm" pb="xs">
+                <Icon {...iconProps} />
+                <Text lh={1}>{t`Official collection`}</Text>
+              </Group>
+            )}
+          </Stack>
+        </SidesheetCard>
+        {collection.entity_id && (
+          <EntityIdCard entityId={collection.entity_id} />
+        )}
+      </Stack>
+    </Sidesheet>
+  );
+};
+
+const EditableDescription = ({
+  collection,
+  handleChangeDescription,
+}: {
+  collection: Collection;
+  handleChangeDescription: (description: string) => void;
+}) => {
+  const description = collection.description;
+  const canWrite = collection.can_write;
+  return (
+    <Box
+      component={EditableText}
+      onChange={handleChangeDescription}
+      initialValue={description}
+      placeholder={
+        !description && !canWrite ? t`No description` : t`Add description`
+      }
+      isDisabled={!canWrite}
+      isOptional
+      isMultiline
+      isMarkdown
+      key={collection.id}
+      pos="relative"
+      left={-5}
+      lh={1.38}
+      tabIndex={0}
+      // For a11y, allow typing to activate the textarea
+      onKeyDown={(e: React.KeyboardEvent) => {
+        if (shouldPassKeyToTextarea(e.key)) {
+          (e.currentTarget as HTMLTextAreaElement).click();
+        }
+      }}
+      onKeyUp={(e: React.KeyboardEvent) => {
+        if (!shouldPassKeyToTextarea(e.key)) {
+          (e.currentTarget as HTMLTextAreaElement).click();
+        }
+      }}
```

Comment:
this is nice!

1) should this be part of the editabletext component?
2) either way, this behavior should be unit tested